mlflow/pyfunc/__init__.py

```diff
@@ -1049,6 +1069,45 @@ def spark_udf(spark, model_uri, result_type="double", env_manager=_EnvManager.LO
 
     model_metadata = Model.load(os.path.join(local_model_path, MLMODEL_FILE_NAME))
 
+    model_output_schema = model_metadata.get_output_schema()
+
+    if result_type is None:
+        if model_output_schema is None:
+            _logger.warning(
+                "No 'result_type' provided for spark_udf and the model metadata does not "
+                "have an output schema. 'result_type' is set to 'double' type as the default "
+                "fallback."
+            )
+            result_type = "double"
+        else:
+            result_type = _infer_spark_udf_return_type(model_output_schema)
+    else:
+        result_type = "boolean" if result_type == "bool" else result_type
+
+    if not isinstance(result_type, SparkDataType):
+        result_type = _parse_datatype_string(result_type)
```

Comment:
Does pyspark have a **_public_** function that works the same as `_parse_datatype_string`?

Reply:
I don't know so far.

Reply:
> Does pyspark have a public function that works the same as _parse_datatype_string?

@HyukjinKwon

Reply:
There isn't a public API for this. One workaround we could do might be `spark.createDataFrame([], result_type).schema`

Reply:
@HyukjinKwon Thanks, I think that's better than `_parse_datatype_string`. I'll file a PR.

Reply:
Filed #8959